# ConflictWiki Data v0.1

We introduce the ConflictWiki dataset, a large collection of [Wikipedia articles on armed conflict](https://en.wikipedia.org/wiki/Category:Conflicts) featuring full text on the document- and section-level. Each conflict is annotated with starting and end date, location, casualty counts, group strengths and conflict outcome. If provided on Wikipedia, entity articles feature auxiliary tabular information on languages, religion, ISO2 code and ideology. 


## Data Directory Structure

The `mappings` data directory contains four subdirectories, `conflict`, `entity`, `conflict_entity` and `network`.

`conflict` contains conflict-related data and information.
`entity` contains entity-related data and information.
`conflict_entity` contains the conflict-entity and entity-entity relationships. 
`network` contains the conflict_entity mappings restructured as nodes and edges. 

If there are multiple files in a directory, they all contain the same content.
We provide most data in multiple data formats (json, csv, pickle). 
The contents of each subdirectory is explained in the following.

```
    |   mappings
        |   conflict
            |   infos / auxiliary information
        |   entity
            |   infos / auxiliary information
        |   conflict_entity
            | ally-enemy pairs
            | conflict_entity_ids
        |   network
                |   aggr_edge_list
                |   node_list
```

### load external libraries

In [1]:
import pandas as pd
import pathlib

## Conflict

contains all information on the conflicts as extracted from the [Wikipedia militarized conflict template](https://en.wikipedia.org/wiki/Template:Infobox_military_conflict).

`conflict_id`, `conflict_name`, `place`,  `date`, `date_start`, `date_end`, `n_belligerents`, `n_entities`, `strength`, `strength_num`, `casualties`, `casualties_num`, `commander`, `commander_num`, `result` tags.<br>
date holds the date as a string, date_start and date_end are extracted from date and are datetime objects; <br>
n_belligerents, n_entities state the number of entities and belligerents involved in the conflict; <br>
strength and casualties estimate group strengths and losses per belligerent (dictionary keys are belligerent indeces); <br>
strength_num and casualties_num are numbers extracted from strength and casualties (dictionary keys are belligerent indeces); <br>
commander lists all military commanders per belligerent (dictionary keys are belligerent indeces); <br>
result is a brief textual summary of the outcome  <br>


In [2]:
conflict_info_file = pathlib.Path().absolute().parent.absolute() / "conflict" / "conflict_info.pkl"
conflict_info = pd.DataFrame.from_dict(pd.read_pickle(conflict_info_file), orient='index')
conflict_info.index.name="conflict_id"
conflict_info

,conflict_name,url,n_belligerents,n_entities,place,date,date_start,date_end,status,casualties,casualties_num,casualties_sum,strength,strength_num,strength_sum,commander,result
conflict_id,,,,,,,,,,,,,,,,,
43063,Operation Anaconda,https://en.wikipedia.org/wiki/Operation_Anaconda,2,16,"Shahi Kot Valley, Paktia Province, Islamic Sta...","March 1-18, 2002",2022-03-01,2021-04-13,,{0: '7 Afghan fighters killed 8 (7 in the Bat...,"{0: [7.0, 8.0, 7.0, 2015.0, 66.0], 1: [500.0, ...","{0: 2103.0, 1: 3382.0}","{0: '30,000', 1: '1,000'}","{0: [30000.0], 1: [1000.0]}","{0: 30000.0, 1: 1000.0}","{0: ['United States', 'Rowan Tink', 'Australia...","Coalition victory, Taliban evacuates but suffe..."
46216,Israeli–Palestinian conflict,https://en.wikipedia.org/wiki/Israeli–Palestin...,2,7,"Middle EastPrimarily in Israel, West Bank, Gaz...",Mid-20th century - presentMain phase: 1964-1993,2021-04-13,2021-04-13,Israeli–Palestinian peace proceslow-level figh...,"{0: '', 1: '', 2: '21,500 casualties (1965–201...","{2: [21500.0, 1965.0, 2013.0, 1946.0, 2012.0, ...",{2: 35439.0},"{0: '', 1: ''}",{},{},"{0: [], 1: []}",None
201936,2003 invasion of Iraq,https://en.wikipedia.org/wiki/2003_invasion_of...,2,17,Iraq,20 March - 1 May 2003,2022-03-20,2003-05-01,None,{0: 'Coalition: 214 killed606 wounded (U.S.)Pe...,"{0: [214.0, 606.0, 24.0, 238.0, 1000.0], 1: [3...","{0: 2082.0, 1: 172996.0, 2: 14769.0}","{0: ': 192,000 personnel: 45,000 troops : 2,00...","{0: [192000.0, 45000.0, 2000.0, 194.0, 70000.0...","{0: 619628.0, 1: 4015761.0}","{0: ['Aleksander Kwa', 'Poland', 'George W. Bu...",Coalition operational success Ba'athist Iraq ...
207031,Battle of Baghdad (2003),https://en.wikipedia.org/wiki/Battle_of_Baghda...,2,3,"Baghdad, Ba'athist Iraq",3-12 April 2003,2021-04-13,2003-04-12,None,"{0: '1,700–2,120 killed (independent estimate)...","{0: [1700.0, 2120.0, 2320.0], 1: [34.0, 1.0, 2...","{0: 6140.0, 1: 54.0}","{0: '45,000', 1: '30,000'}","{0: [45000.0], 1: [30000.0]}","{0: 45000.0, 1: 30000.0}","{0: ['Ra'ad al-Hamdani', 'Sayf al-Din al-Rawi'...",Decisive Coalition victory Fall of Ba'ath Par...
312905,Lebanese Civil War,https://en.wikipedia.org/wiki/Lebanese_Civil_War,4,28,Lebanon,13 April 1975 - 13 October 1990 Battle of Sido...,1975-04-13,1990-10-13,None,{},{},{},"{0: '', 1: '', 2: ' 25,000 troops (1976)', 3: ...","{2: [25000.0, 1976.0], 3: [1200.0, 1000.0, 100...","{2: 26976.0, 3: 4600.0}","{0: ['Death by natural causes', 'Etienne Saqr'...",Taif Agreement Christian 55:45 ascendancy repl...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
66384676,Battle of Konduga (2015),https://en.wikipedia.org/wiki/Battle_of_Kondug...,2,2,"Konduga, Nigeria",2 March 2015,2015-03-02,None,None,"{0: '73 insurgents', 1: '1 soldier'}","{0: [73.0], 1: [1.0]}","{0: 73.0, 1: 1.0}","{0: '~150 insurgents', 1: 'Unknown number of s...","{0: [150.0], 1: [1.0]}","{0: 150.0, 1: 1.0}",{},Nigerian victory
66391098,Battle of Damboa,https://en.wikipedia.org/wiki/Battle_of_Damboa,2,2,"Damboa, Nigeria",October 6 2013,2013-10-06,None,None,"{0: '15 killed', 1: 'none', 2: '7 civilians ki...","{0: [15.0], 2: [7.0]}","{0: 15.0, 2: 7.0}","{0: 'Unknown', 1: 'Unknown'}",{},{},{},Nigerian victory
66527448,Siege of Qamishli and Al-Hasakah,https://en.wikipedia.org/wiki/Siege_of_Qamishl...,2,2,"Qamishli and Al-Hasakah, Syria",10 January - 2 February 202,2022-01-10,2022-02-02,None,"{0: 'None', 1: '1 killedhttps://www.syriahr.co...","{1: [1.0, 203176.0, 7.0, 203176.0, 3.0], 2: [6...","{1: 406363.0, 2: 203182.0}","{0: '', 1: '', 2: ''}",{},{},{},Siege lifted


## Entity

contains all information on the entities as extracted from the [Wikipedia article info box](https://en.wikipedia.org/wiki/Mali).

`entity_id`, `entity_name`, `num_conflicts`, `iso`, `language`, `religion`, `ideology` <br>
num_conflicts gives the number of conflicts, the entity is involved in; <br>

In [9]:
entity_info_file = pathlib.Path().absolute().parent.absolute() / "entity" / "entity_info.pkl"
entity_info = pd.DataFrame.from_dict(pd.read_pickle(entity_info_file), orient='index')
entity_info.index.name="entity_id"
entity_info

,entity_name,url,num_conflicts,iso,language,ideology,religion
entity_id,,,,,,,
4689264,Australia,https://en.wikipedia.org/wiki/Australia,32,AU,"[National language, English, Australian Englis...",[],[]
3434750,United States,https://en.wikipedia.org/wiki/United_States,301,US,"[Spanish, Algonquian languages, Cherokee langu...",[],[Religion in the United States]
5042916,Canada,https://en.wikipedia.org/wiki/Canada,42,CA,"[English, Canadian French, Canadian English, F...",[],[]
76972,Denmark,https://en.wikipedia.org/wiki/Denmark,24,DK,"[Danish, Danish language]",[],"[Church of the Faroe Islands, Greenland, No re..."
11867,Germany,https://en.wikipedia.org/wiki/Germany,20,EU,[European Charter for Regional or Minority Lan...,[],[]
...,...,...,...,...,...,...,...
38762388,Sahrawi Arab Democratic Republic,https://en.wikipedia.org/wiki/Sahrawi_Arab_Dem...,2,<!,"[Spanish, Arabic, Spanish language, Modern Sta...",[],[Islam]
4853315,Gumuz people,https://en.wikipedia.org/wiki/Gumuz_people,1,[],[],[],[]
6223534,Oromo Liberation Front,https://en.wikipedia.org/wiki/Oromo_Liberation...,1,[],[],"[Ethnic nationalism, Secularism, Oromo nationa...",[]


## Conflict_Entity

##### conflict_entity_id

Contains the ids of all conflicts and the ids of all involved entities partitioned into belligerents.
In the examplatory conflict below, we have three belligerents, of which the first two are formed by two entities each 
and the third belligerent consists of only one entity

```
{conflict_id: [[entity_id, entity_id, entity_id, entity_id],[entity_id]]}
(1220919: [[1576797, 31717, 4887, 3434750], [40596311]])
```

In [10]:
conflict_entity_id_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "conflict_entity" / "conflict_entity_id.pkl"
#conflict_entity_id = pd.read_pickle(conflict_entity_id_file)
conflict_entity_id= pd.DataFrame.from_dict(pd.read_pickle(conflict_entity_id_file), orient='index', columns = ["belligerent_1", "belligerent_2", "belligerent_3", "belligerent_4"])
conflict_entity_id.index.name="conflict_id"
conflict_entity_id

,belligerent_1,belligerent_2,belligerent_3,belligerent_4
conflict_id,,,,
22738,"[27019, 1737554, 17207794, 4913064, 23235, 318...","[103100, 51673, 737, 30635, 27358, 21462079, 3...",None,None
43063,"[21241, 291717, 4913064, 31717, 11125639, 737,...","[389363, 737, 4512092]",None,None
46216,[9282173],"[24093, 12047, 24324, 13913, 53185055, 13093253]",None,None
106346,"[1113713, 23235, 3607937, 31769, 3434750]",[8089426],None,None
201936,"[899044, 22936, 31717, 679693, 7515928, 256597...","[7515849, 26215470, 285632, 2185, 199866, 17633]",None,None
...,...,...,...,...
64786838,[469639],"[43667081, 298010]",None,None
65268074,"[12593188, 171166, 5217236, 14833748]",[66890],None,None
65431221,"[11125639, 73374, 746, 8862873]","[214413, 3057255, 25391, 44596915, 10918072]",None,None


##### ally_enemy_pairs

Contains the ids of entity pairs, a conflict id and the relationship of the entities in the given conflict as displayed below

```
((entity_pair), conflict_id, entity_relationship)
((27019, 103100), 22738, "enemies")

In [10]:
ally_enemy_pairs_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "conflict_entity" / "ally_enemy_pairs.pkl"
#ally_enemy_pairs = pd.read_pickle(ally_enemy_pairs_file)
ally_enemy_pairs= pd.DataFrame(pd.read_pickle(ally_enemy_pairs_file),columns = ["entity_pair", "conflict_id", "entity_relationship"])
ally_enemy_pairs.index.name="entity_pair_id"
ally_enemy_pairs

,entity_pair,conflict_id,entity_relationship
entity_pair_id,,,
0,"(27019, 103100)",22738,enemies
1,"(27019, 51673)",22738,enemies
2,"(27019, 737)",22738,enemies
3,"(27019, 30635)",22738,enemies
4,"(27019, 27358)",22738,enemies
...,...,...,...
17066,"(25391, 44596915)",66219979,enemies
17067,"(25391, 10918072)",66219979,enemies
17068,"(44596915, 10918072)",66219979,enemies


## Network

Contains the precomputed, aggregated network, where nodes are entities and edges are all conflicts between any two respective entities aggregated.

###### aggr_edge_list

List of edges of shape (node_id 1, node_id 2, edge attributes)

```
(entity_id 1, entity_id 2, conflict attributes
27019,	103100,	{'label': 'enemies', 'label_discrete': 0, 'label_continuous': -1, 'n_conflicts': 1, 'conflict_ids': [22738], 'conflict_names': ['Operation Enduring Freedom']}
```

In [11]:
aggr_edge_list_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "network" / "aggr_edge_list.pkl"
aggr_edge_list = pd.DataFrame(pd.read_pickle(aggr_edge_list_file),columns = ["entity_id_1", "entity_id_1", "conflict_attributes"])
aggr_edge_list.index.name="edge"
aggr_edge_list

,entity_id_1,entity_id_1,conflict_attributes
edge,,,
0,27019,103100,"{'label': 'enemies', 'label_discrete': 0, 'lab..."
1,27019,51673,"{'label': 'enemies', 'label_discrete': 0, 'lab..."
2,737,27019,"{'label': 'enemies', 'label_discrete': 0, 'lab..."
3,27019,30635,"{'label': 'enemies', 'label_discrete': 0, 'lab..."
4,27019,27358,"{'label': 'allies', 'label_discrete': 1, 'labe..."
...,...,...,...
10832,214413,44596915,"{'label': 'allies', 'label_discrete': 1, 'labe..."
10833,214413,10918072,"{'label': 'allies', 'label_discrete': 1, 'labe..."
10834,25391,3057255,"{'label': 'allies', 'label_discrete': 1, 'labe..."


###### node_list

List of nodes (node_id, node attributes)

```
(entity_id, entity attributes
3343, {'name': 'Belgium'}
```

In [12]:
node_list_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "network" / "node_list.pkl"
node_list= pd.DataFrame(pd.read_pickle(node_list_file),columns = ["entity_id", "entity_attributes"])
node_list.index.name="node"
node_list

,entity_id,entity_attributes
node,,
0,358,{'name': 'Algeria'}
1,701,{'name': 'Angola'}
2,737,{'name': 'Afghanistan'}
3,738,{'name': 'Albania'}
4,746,{'name': 'Azerbaijan'}
...,...,...
711,38264234,{'name': 'European Union Training Mission in M...
712,40479968,{'name': 'United Nations Force Intervention Br...
713,43112164,{'name': 'European Union Training Mission in S...
